cvmfs/publish/cmd_commit.cc

```diff
@@ -0,0 +1,103 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include "cvmfs_config.h"
+#include "cmd_commit.h"
+
+#include <errno.h>
+
+#include <ctime>
+#include <string>
+#include <vector>
+
+#include "logging.h"
+#include "publish/cmd_util.h"
+#include "publish/except.h"
+#include "publish/repository.h"
+#include "publish/settings.h"
+#include "util/posix.h"
+#include "util/string.h"
+#include "whitelist.h"
+
+namespace publish {
+
+int CmdCommit::Main(const Options &options) {
+  // Repository name and lease path are submitted as a single argument
+  // for historical reasons
+  std::string fqrn;
+  std::string lease_path;
+
+  if (!options.plain_args().empty()) {
+    std::vector<std::string> tokens =
+      SplitString(options.plain_args()[0].value_str, '/', 2);
+    fqrn = tokens[0];
+    if (tokens.size() == 2)
+      lease_path = MakeCanonicalPath(tokens[1]);
+  }
+
+  std::string session_dir = Env::GetEnterSessionDir();
+  std::string config_tmp = session_dir + "/tmp.conf";
+  std::string config;
+  int fd_config = open(config_tmp.c_str(), O_RDONLY);
+  SafeReadToString(fd_config, &config);
+
+  SettingsBuilder builder;
+  builder.config_path_ = config;
+
+  UniquePtr<SettingsPublisher> settings;
+  try {
+    settings = builder.CreateSettingsPublisher(fqrn, true /* needs_managed */);
+  } catch (const EPublish &e) {
+    if (e.failure() == EPublish::kFailRepositoryNotFound) {
+      LogCvmfs(kLogCvmfs, kLogStderr | kLogSyslogErr, "CernVM-FS error: %s",
+               e.msg().c_str());
+      return 1;
+    }
+    throw;
+  }
+
+  if (!SwitchCredentials(settings->owner_uid(), settings->owner_gid(),
+                         false /* temporarily */))
+  {
+    throw EPublish("No write permission to repository");
+  }
+
+  FileSystemInfo fs_info = GetFileSystemInfo("/cvmfs");
+  if (fs_info.type == kFsTypeAutofs)
+    throw EPublish("Autofs on /cvmfs has to be disabled");
+
+  settings->GetTransaction()->SetLeasePath(lease_path);
```

Comment:
The lease path should not be required here. At this point, the command will use the already acquired lease token.